src/service/db/dashboards/mod.rs

```diff
@@ -289,12 +81,42 @@ pub(crate) async fn delete(
     dashboard_id: &str,
     folder: &str,
 ) -> Result<(), anyhow::Error> {
-    let key = format!("/dashboard/{org_id}/{folder}/{dashboard_id}");
-    Ok(db::delete(&key, false, db::NO_NEED_WATCH, None).await?)
+    dashboards::delete(org_id, folder, dashboard_id).await?;
+    Ok(())
 }
 
 #[tracing::instrument]
 pub async fn reset() -> Result<(), anyhow::Error> {
-    let key = "/dashboard/";
-    Ok(db::delete(key, true, db::NO_NEED_WATCH, None).await?)
+    dashboards::delete_all().await?;
+    Ok(())
+}
+
+/// Parses the bytes into a dashboard with the given version.
+pub(crate) fn parse_dashboard(
+    bytes: web::Bytes,
+    version: i32,
+) -> Result<Dashboard, serde_json::Error> {
+    let dash = match version {
+        1 => {
+            let inner: v1::Dashboard = json::from_slice(&bytes)?;
+            inner.into()
+        }
+        2 => {
+            let inner: v2::Dashboard = json::from_slice(&bytes)?;
+            inner.into()
+        }
+        3 => {
+            let inner: v3::Dashboard = json::from_slice(&bytes)?;
+            inner.into()
+        }
+        4 => {
+            let inner: v4::Dashboard = json::from_slice(&bytes)?;
+            inner.into()
+        }
+        _ => {
+            let inner: v5::Dashboard = json::from_slice(&bytes)?;
+            inner.into()
+        }
+    };
+    Ok(dash)
 }
```

Comment:
Admittedly, I don't think this is a good place for this function - but at this moment I don't think there's really a better place.

Since this function is specifically concerned with transforming from the representation of a dashboard in an HTTP request into our domain model of a dashboard, I think that this logic ideally should live inside our `http` module. However at the moment HTTP-specific logic bleeds into `/service/dashboards` and `/service/db/dashboards` - so this logic is needed here. 

As a follow up to cleaning up some tech debt, I would like to put up some PRs that split and create a harder division between HTTP request-handling logic VS service layer domain logic. I can move this logic into the `http` module when I work on that PR.